packages/editor/src/utils/with-history/test/index.js

```diff
@@ -167,6 +167,62 @@ describe( 'withHistory', () => {
 		} );
 	} );
 
+	it( 'should overwrite present state with option.shouldOverwriteState right after ignored action', () => {
+		const complexCounter = ( state = { count: 0 }, action ) => {
+			if ( action.type === 'INCREMENT' ) {
+				return {
+					...state,
+					count: state.count + 1,
+				};
+			} else if ( action.type === 'IGNORE' ) {
+				return {
+					...state,
+					ignore: action.content,
+				};
+			}
+
+			return state;
+		};
+
+		const reducer = withHistory( {
+			shouldOverwriteState: ( action, previousAction ) =>
+				previousAction && action.type === previousAction.type,
+			ignoreTypes: [ 'IGNORE' ],
+		} )( complexCounter );
+
+		let state;
```

Comment:
Any reason not to do:

```js
let state = reducer( undefined, {} );
state = reducer( state, { type: 'INCREMENT' } );
```

Reply:
Just copied another test tbh. 😅

Reply:
![it-was-like-that-when-i-got-here](https://user-images.githubusercontent.com/90871/47912671-6cb18e80-de91-11e8-96d3-09a361df8abf.jpg)